lib/widgets/rectangular_inset_shadow_box.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/widgets.dart';
```

Comment:
nit: generally we've followed the Flutter upstream tree in grouping related functionality together into source files, rather than atomizing each class into its own tiny file.

One consequence is that the file names are shorter and more general than the class names.

Here there might not be any existing other code for this to share a file with, but we can still give it a bit of a broader, shorter name. Say `inset_shadow.dart`.